### PR TITLE
docs: Revamped Deployment Chart Doc

### PR DESCRIPTION
**Pull request by ashokdevtron**
1. Included steps for **app-values.yaml** and **release-values.yaml** in prerequisites
2. Added video tutorial
3. Updated existing snapshots
4. Restructured doc with additional info
5. Added ready to use content for yaml files
6. Added one troubleshooting item in FAQ

### CODE REVIEW

File: docs/FAQs/devtron-troubleshoot.md [hunk 1]
```diff
@@ -613,3 +613,11 @@ Follow the below steps if you are getting the above error:
 
   ![](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/devtron-troubleshooting/chart-sync.jpg)
 
+#### 30. The Advanced (YAML) and Basic (GUI) sections are appearing blank in the Base Deployment Template of the application.
+
+![](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/global-configurations/deployment-charts/empty-values.jpg)
+
+If you have uploaded your deployment chart which doesn't have an [app-values.yaml](../user-guide/global-configurations/deployment-charts.md#3-add-app-values.yaml) file and then select such chart as the [Base Deployment Template](../creating-application/deployment-template.md) of your application, the **Basic (GUI)** and **Advanced (YAML)** sections will display blank.
```
Comment (abhibhaw) [inline docs/FAQs/devtron-troubleshoot.md:620]: `and then select such chart` this sounds a bit unease

File: docs/user-guide/global-configurations/deployment-charts.md [hunk 2]
```diff
@@ -1,70 +1,94 @@
...
+| `Description` | Give a description to your chart (Optional). |
 
-Please see the following example:
+{% hint style="info" %}
+### Example of Chart.yaml
+[Click here](https://devtron-public-asset.s3.us-east-2.amazonaws.com/custom-charts/chart-yaml-file.png) to view a sample `Chart.yaml` file.
+{% endhint %}
 
-![Chart.yaml file](https://devtron-public-asset.s3.us-east-2.amazonaws.com/custom-charts/chart-yaml-file.png)
+### 2. Create an Image Descriptor Template File
```
Comment (abhibhaw) [inline docs/user-guide/global-configurations/deployment-charts.md:41]: why an image descriptor is required?

File: docs/user-guide/global-configurations/deployment-charts.md [hunk 3]
```diff
@@ -1,70 +1,94 @@
...
+        "app": "{{.App}}",
+        "env": "{{.Env}}",
+        "appMetrics": {{.AppMetrics}}
+    }
+    ```
+
+    The above code is a GO template file that produces a valid JSON file upon rendering. The values from the CD deployment pipeline are injected at the placeholders specified in `.image_descriptor_template.json`.
+
+
+    All the placeholders are optional. Let's say you wish to create a template file that allows Devtron to render only the repository name and the tag from the CI/CD pipeline you created, edit the `.image_descriptor_template.json` file as follows:
```
Comment (abhibhaw) [inline docs/user-guide/global-configurations/deployment-charts.md:75]: render is wrong word here

File: docs/user-guide/global-configurations/deployment-charts.md [hunk 4]
```diff
@@ -73,107 +97,145 @@ You can use the following variables in the helm template (all the placeholders a
...
->     "image": {
->	        "repository": "{{.Name}}",
->	        "tag": "{{.Tag}}"
->     }
-> }
-> ```
+### 3. Add app-values.yaml
 
-### 3. Package the custom chart in the `*.tgz` format
+In the root directory of your chart, Devtron expects an `app-values.yaml` file and validates whether the content of `values.yaml` file is present in `app-values.yaml` file or not.
```
Comment (abhibhaw) [inline docs/user-guide/global-configurations/deployment-charts.md:104]: why is app-values.yaml required? why my values.yaml should be repeated in app-values.yaml?

File: docs/user-guide/global-configurations/deployment-charts.md [hunk 5]
```diff
@@ -73,107 +97,145 @@ You can use the following variables in the helm template (all the placeholders a
...
 
-> Before you begin, ensure that your helm chart includes both `Chart.yaml` (with `name` and `version` fields) and `.image_descriptor_template.json` files.
+You may use the following command:
 
-The helm chart to be uploaded must be packaged as a versioned archive file in the format `<helm-chart-name>-vx.x.x.tgz`.
+```bash
+cp values.yaml app-values.yaml
+```
+
+### 4. Add release-values.yaml
```
Comment (abhibhaw) [inline docs/user-guide/global-configurations/deployment-charts.md:112]: why to add release-values.yaml?

File: docs/user-guide/global-configurations/deployment-charts.md [hunk 6]
```diff
@@ -73,107 +97,145 @@ You can use the following variables in the helm template (all the placeholders a
...
+
+In the root directory of your chart create a file named `release-values.yaml`. You may use the following command:
 
+```bash
+touch release-values.yaml
 ```
+
+Paste the following content in `release-values.yaml` file:
```
Comment (abhibhaw) [inline docs/user-guide/global-configurations/deployment-charts.md:120]: why this specfic content?

File: docs/user-guide/global-configurations/deployment-charts.md [hunk 7]
```diff
@@ -73,107 +97,145 @@ You can use the following variables in the helm template (all the placeholders a
...
-| **New version detected** | You are uploading a newer version of an existing chart | Enter a **Description** and select **Save** to continue uploading, or **Cancel upload**. |
-| **Already exists** | There already exists a chart with the same version. | <ul><li>Edit the version and re-upload the same chart using **Upload another chart**.</li><li>Upload a new chart with a new name using  **Upload another chart**.</li><li>**Cancel upload**.</li></ul> |
+### Validation Checks
 
-![Chart validated](https://devtron-public-asset.s3.us-east-2.amazonaws.com/custom-charts/List+-+Empty-2.png)
+In the uploading process, your file will be validated against the following criteria:
 
-![Unsupported template](https://devtron-public-asset.s3.us-east-2.amazonaws.com/custom-charts/List+-+Empty.png)
+- Supported archive template should be in `*.tgz` format.
+- ConfigMap/Secret template should be the same as that of our [reference chart](https://github.com/devtron-labs/devtron/tree/main/scripts/devtron-reference-helm-charts/reference-chart_4-14-0).
```
Comment (abhibhaw) [inline docs/user-guide/global-configurations/deployment-charts.md:191]: I doubt if we do validate the cm/secret template

**Review by abhibhaw — CHANGES_REQUESTED** [review]

File: docs/FAQs/devtron-troubleshoot.md (hunk 1 above)
Comment (ashokdevtron) [reply docs/FAQs/devtron-troubleshoot.md:620]: Fixed

File: docs/user-guide/global-configurations/deployment-charts.md (hunk 2 above)
Comment (ashokdevtron) [reply docs/user-guide/global-configurations/deployment-charts.md:41]: Was already explained few lines below, but enhanced it now

File: docs/user-guide/global-configurations/deployment-charts.md (hunk 3 above)
Comment (ashokdevtron) [reply docs/user-guide/global-configurations/deployment-charts.md:75]: Fixed

File: docs/user-guide/global-configurations/deployment-charts.md (hunk 4 above)
Comment (ashokdevtron) [reply docs/user-guide/global-configurations/deployment-charts.md:104]: Explained it in the new PR

File: docs/user-guide/global-configurations/deployment-charts.md (hunk 7 above)
Comment (ashokdevtron) [reply docs/user-guide/global-configurations/deployment-charts.md:191]: Correct, we don't validate

**Pull request closed without merging**